doc/src/manual/style-guide.md

```diff
@@ -164,6 +164,62 @@ uses (e.g. `a[i]::Int`) than to try to pack many alternatives into one type.
 If a function name requires multiple words, consider whether it might represent more than one
 concept and might be better split into pieces.
 
+## Write functions that use similar argument precedence as Julia's `base/`
+
+As a general rule, the Base library uses the following order of arguments to functions,
+as applicable:
+
+1. **Function argument**.
+   This permits the use of `do` blocks for passing multiline anonymous functions.
+
+2. **I/O stream**.
+   Specifying the `IO` object first permits passing the function to functions such as
+   [`sprint`](@ref), e.g. `sprint(show, x)`.
+
+3. **Input being mutated**.
+   In [`fill!(x, v)`](@ref fill!), `x` is the object being mutated, and it appears before
+   the value to be inserted into `x`.
+
+4. **Type**.
+   Passing a type typically means that the output will have the given type.
+   In [`read(io, String)`](@ref read), the `IO` argument appears before the type.
```

Comment:
Do we have a better example for this? Where the type actually is the first argument? Perhaps `parse(Int, "123")` or something?

Reply:
`parse` is a good example, I'll add that as well, but I thought `read` was apt here because it shows that `IO` comes before types.